src/screens/nano/NanoContractDetail.js

```diff
@@ -0,0 +1,205 @@
+    return {
+      nanoContracts: state.nanoContracts,
+      blueprintsData: state.blueprintsData,
+      tokenMetadata: state.tokenMetadata
+    }
+  });
+
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const { nc_id: ncId } = useParams();
+  const nc = nanoContracts[ncId];
+  let blueprintInformationAux = blueprintsData[nc.blueprintId];
+
+  // loading {boolean} Bool to show/hide loading element
+  const [loading, setLoading] = useState(true);
+  // data {Object} Nano contract loaded data
+  const [data, setData] = useState(null);
+  // blueprintInformation {Object} Blueprint information data
+  const [blueprintInformation, setBlueprintInformation] = useState(blueprintInformationAux);
+  // errorMessage {string} Message to show when error happens on the form
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    loadData();
+  }, []);
+
+  /**
+   * Method executed when link to execute a method is clicked
+   *
+   * @param {Object} e Event emitted by the link clicked
+   * @param {String} method Method to be executed
+   */
+  const executeMethod = (e, method) => {
+    e.preventDefault();
+    navigate('/nano_contract/execute_method/', {
+      state: {
+        method,
+        ncId,
+      },
+    });
+  }
+
+  /**
+   * Method executed when link to change the address is clicked
+   *
+   * @param {Object} e Event emitted by the link clicked
+   */
+  const changeAddress = (e) => {
+    e.preventDefault();
+    context.showModal(MODAL_TYPES.NANOCONTRACT_CHANGE_ADDRESS, {
+      nanoContractID: ncId
+    });
+  }
+
+  const loadData = async () => {
+    await loadBlueprintInformation();
+    await loadNCData();
+  }
+
+  const loadBlueprintInformation = async () => {
+    if (blueprintInformationAux) {
+      return;
+    }
+
+    setLoading(true);
+    try {
+      const blueprintInformationResponse = await hathorLib.ncApi.getBlueprintInformation(nc.blueprintId);
+      // We need this blueprint information response to call the following get state
+      // The set state is not sync, so we need to store it in a common variable to be used in the next call
+      blueprintInformationAux = blueprintInformationResponse;
+      setBlueprintInformation(blueprintInformationResponse);
+      // Store in redux, so it can be reused by other nano contracts
+      dispatch(addBlueprintInformation(blueprintInformationResponse));
+    } catch(e) {
+      // Error in request
+      setErrorMessage(t`Error getting blueprint details.`);
+    };
+  }
+
+  const loadNCData = async () => {
+    setLoading(true);
+    setData(null);
+    try {
+      const state = await hathorLib.ncApi.getNanoContractState(ncId, Object.keys(blueprintInformationAux.attributes), [], []);
+      setData(state);
+    } catch(e) {
+      // Error in request
+      setErrorMessage(t`Error getting nano contract state.`);
+    } finally {
+      setLoading(false);
+    };
+  }
```

Comment:
```suggestion
  const loadData = async () => {
    setLoading(true);
    try {
      await loadBlueprintInformation();
      await loadNCData();
    } finally {
      setLoading(false);
    }
  }

  const loadBlueprintInformation = async () => {
    if (blueprintInformationAux) {
      return;
    }

    try {
      const blueprintInformationResponse = await hathorLib.ncApi.getBlueprintInformation(nc.blueprintId);
      // We need this blueprint information response to call the following get state
      // The set state is not sync, so we need to store it in a common variable to be used in the next call
      blueprintInformationAux = blueprintInformationResponse;
      setBlueprintInformation(blueprintInformationResponse);
      // Store in redux, so it can be reused by other nano contracts
      dispatch(addBlueprintInformation(blueprintInformationResponse));
    } catch(e) {
      // Error in request
      setErrorMessage(t`Error getting blueprint details.`);
    };
  }

  const loadNCData = async () => {
    setData(null);
    try {
      const state = await hathorLib.ncApi.getNanoContractState(ncId, Object.keys(blueprintInformationAux.attributes), [], []);
      setData(state);
    } catch(e) {
      // Error in request
      setErrorMessage(t`Error getting nano contract state.`);
    };
  }
```

Reply:
Much better, thanks https://github.com/HathorNetwork/hathor-wallet/pull/623/commits/9586f5479bc223a3bcc22f4b17237894162b8759